### ISSUE TITLE
typings: setKeyboardHandler does not indicate that it accepts null

### ISSUE THREAD

**Issue by shepmaster** [ajaxorg/ace#4801]
### Describe the bug

The documentation (emphasis mine) indicates that the argument cannot be `null`:

> setKeyboardHandler(**String** keyboardHandler)
> Sets a new key handler, such as "vim" or "windows".
> Arguments
> keyboardHandler	String	
> **Required**. The new key handler

The TypeScript typings agree:

https://github.com/ajaxorg/ace/blob/7a4f9e29b90a5f923f89a28dda678c728e3f94f6/ace.d.ts#L772

However, `null` is required in order to set the implicit "Ace" keybindings, as shown on the kitchen sink:

```html
<button value="null" aria-pressed="true" ace_selected_button="true">Ace</button>
```

### Expected Behavior

The TypeScript typings should not produce an error when passing `null` to  `setKeyboardHandler`.

### Current Behavior

```
TS2345: Argument of type 'string | null' is not assignable to parameter of type 'string'.
```

or

```
  Overload 1 of 2, '(keyboardHandler: string, callback?: (() => void) | undefined): void', gave the following error.
    Argument of type 'null' is not assignable to parameter of type 'string'.
  Overload 2 of 2, '(keyboardHandler: KeyboardHandler): void', gave the following error.
    Argument of type 'null' is not assignable to parameter of type 'KeyboardHandler'.
```

### Reproduction Steps

```
mkdir repro
cd repro
yarn init repro
yarn add typescript ace-builds
cat > index.ts <<EOF
type AceEditor = import('ace-builds').Ace.Editor;

function demo(ace: AceEditor, handler: string | null) {
  ace.setKeyboardHandler(null);
}
EOF
yarn run tsc --noEmit --strict index.ts
```

### Possible Solution

Correct the docs and typings to allow for `null`. It's also possible I've misunderstood the signature; in that case the documentation should be updated to indicate how to select the Ace keybinding.

### Ace Version

1.5.2

**Comment by andrewnester** [comment]
It's best to use `editor.setOption("keybindings", "default")` to set default keybindings. `setKeyboardHandler ` is technically correct even though indeed due to internal implementation of it it's possible to pass `null` but we don't allow this from public API perspective.

The documentation needs to be updated though

**Comment by shepmaster** [comment]
It appears that if I use `ace`, any future keystrokes result in

```
ace.js:4202 Uncaught TypeError: this.$handlers[i].handleKeyboard is not a function
    at KeyBinding.$callKeyboardHandlers (ace.js:4202:1)
    at KeyBinding.onTextInput (ace.js:4238:1)
    at Editor.onTextInput (ace.js:13362:1)
    at sendText (ace.js:1717:1)
    at HTMLTextAreaElement.onInput (ace.js:1743:1)
```

**Comment by shepmaster** [comment]
> use `editor.setOption("keybindings", "default")`

Did you mean `"keyboardHandler"`? If I use `"keybindings"` I get a warning:

```
misspelled option "keybindings"
warn @ ace.js:3602
setOption @ ace.js:3579
```

**Comment by andrewnester** [comment]
Sorry for the confusion. So what I meant initially is you can set keybindings like this, see code here
https://github.com/ajaxorg/ace/blob/d95725983b363a616c584237013dfd36eaadbba4/lib/ace/ext/textarea.js#L304-L314
But this applies for textarea ext only

Generally `editor.setOption("keyboardHandler", null)` should work

**Comment by shepmaster** [comment]
> Generally `editor.setOption("keyboardHandler", null)` should work

Ok, then this issue should become "`editor.setOption('keyboardHandler',` doesn't indicate it accepts null" :-)

The original reproduction can be updated with 

```ts
type AceEditor = import('ace-builds').Ace.Editor;

function demo(ace: AceEditor, handler: string | null) {
  ace.setOption('keyboardHandler', handler);
}
```

```
index.ts:4:36 - error TS2345: Argument of type 'string | null' is not assignable to parameter of type 'string'.
  Type 'null' is not assignable to type 'string'.

4   ace.setOption('keyboardHandler', handler);
                                     ~~~~~~~
```

**Comment by andrewnester** [comment]
Released in 1.9 version
